packages/ui/src/components/postCollectionViews/CarouselPostCollectionView.tsx

```diff
@@ -0,0 +1,149 @@
+import { JSONValue } from '@tloncorp/shared';
+import {
+  forwardRef,
+  useCallback,
+  useImperativeHandle,
+  useMemo,
+  useRef,
+} from 'react';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import { Theme, View, XStack, YStack } from 'tamagui';
+import { LinearGradient } from 'tamagui/linear-gradient';
+
+import { usePostCollectionContextUnsafelyUnwrapped } from '../../contexts/postCollection';
+import AuthorRow from '../AuthorRow';
+import { Carousel, CarouselRef } from '../Carousel';
+import { Icon } from '../Icon';
+import { Text } from '../TextV2';
+import { IPostCollectionView } from './shared';
+
+function BaseCarouselPostCollection() {
+  const {
+    hasNewerPosts,
+    hasOlderPosts,
+    onScrollEndReached,
+    onScrollStartReached,
+    posts,
+    collectionConfiguration,
+    PostView,
+  } = usePostCollectionContextUnsafelyUnwrapped();
+
+  const carouselRef = useRef<CarouselRef>(null);
+  const handleIndexChange = useCallback(
+    (newIndex: number) => {
+      if (newIndex === 0 && hasNewerPosts) {
+        onScrollStartReached?.();
+      }
+      if (posts && newIndex === posts.length - 1 && hasOlderPosts) {
+        onScrollEndReached?.();
+      }
+    },
+    [
+      hasNewerPosts,
+      hasOlderPosts,
+      onScrollEndReached,
+      onScrollStartReached,
+      posts,
+    ]
+  );
+
+  const scrollDirection = useMemo(() => {
+    try {
+      return JSONValue.asString(
+        collectionConfiguration?.scrollDirection,
+        'vertical'
+      ) as 'vertical' | 'horizontal';
+    } catch (_) {
+      // ignore
+      return 'vertical';
+    }
+  }, [collectionConfiguration]);
+
+  return (
+    <View alignItems="stretch" justifyContent="center" flex={1}>
+      <Carousel
+        ref={carouselRef}
+        flex={1}
+        scrollDirection={scrollDirection}
+        onVisibleIndexChange={handleIndexChange}
+      >
+        {posts?.map((post) => (
+          <Carousel.Item justifyContent="center" key={post.id}>
+            <XStack flex={1} backgroundColor="white">
```

Comment:
probably better to use a theme color instead of `white` for dark mode support? (is this background visible at all, maybe in overscroll?)

Reply:
Oops yeah missed that -- it's visible when trying to use non-media posts with this collection.